instructor/ui/instructor.js

```diff
@@ -631,48 +663,107 @@ $(function() {
     /* Get name of the sample from the row itself */
     var sample_name = $("#" + row_id).text().replace(/(\n *)+/g, "");
     $('.scb_ab_f_sample_name').text(sample_name);
     var instance_pk = row_id.match(/(\d+)$/)[0];
-    var sample_treatment  = $(this).data('sample_treatment');
+    var sample_treatment  = $(this).data('sample_treatment'),
+        group_id = $(this).data('group_id') ? $(this).data('group_id'): null,
+        analysis = sample_treatment.split(',')[0];
+
     $('.scb_ab_f_treatment_text').text(sample_treatment);
     $('.scb_ab_s_analyze_dialog').css('visibility', 'visible');
     $('.scb_ab_f_save_image').data({
-      'pk': instance_pk,
-      'filter_group_id': filter_group_id
+        'pk': instance_pk,
+        'group_id': group_id,
+        'analysis': analysis,
     });
-    addImageFormHandler();
+    addSelectedImages(); // Add already saved images to the selected area
+  });
 
+  $(function(){
+   addSelectedImages(); // Add already saved images to the selected area on the page loading
   });
+
+  /* Remove remove selection from the images and revert all unsaved images to the image bank */
+  function clearSelectedFrame() {
+      var reselectImages = $(".scb_ab_s_small_image_selected");
+      reselectImages.each(function () {
+          $(this).removeClass("scb_ab_s_small_image_selected");
+          $(".scb_ab_s_image_bank").append($(this));
+      });
+  }
+
+  /* Fulfil filter area by chosen image in the Fluorescent image sececting flow*/
+  function fulfilFilter(group_id, chosenImages) {
+      for (var filter in filterMap) {
+          var filterName = filterMap[filter],
+              $filter = $(".scb_f_image_filter_{}".replace('{}', filterName)),
+              filterImage = chosenImages[filterName];
+          $filter.data("filter_group", filter + "-" + group_id);
+          if (filterImage) {
+            shiftImages(filterImage, $filter)
+          } else {
+            $filter.text("{} filter".replace('{}', filterName));
+          }
+          $(".scb_ab_s_select_box").append($filter);
+      }
+  }
+
+  /* Clarify chosen image from the images in the bank and move it to the target element*/
+  function shiftImages(chosenImage, elementToAppend) {
+      var imageUrlList = chosenImage.attr("src").split('/'),
+          imageName = imageUrlList[imageUrlList.length - 1],
+          chosenImages = $(".scb_ab_s_image_bank img[src$='{}']".replace("{}", imageName));
+      chosenImages.addClass('scb_ab_s_small_image_selected');
+      elementToAppend.append(chosenImages);
+  };
+
+
+  var filterMap = ['red', 'blue', 'green', 'merge'];
+
+  /* Add saved images to the selected field on the reopening Upload Image(s) tab */
+  function addSelectedImages() {
+      clearSelectedFrame();
+      var chosenImages = {},
+          group_id = $(".scb_ab_f_save_image").data('group_id'),
+          analysis = $(".scb_ab_f_save_image").data('analysis'),
+          filtersFrames = $(".scb_f_image_filter");
```

Comment:
To follow the convention in this file of appending JQuery object vars with $, this should be named `$filtersFrames`.

Reply:
Good catch, thanks. fixed